onboard/catkin_ws/src/acoustics/scripts/acoustics_wrapper.py

```diff
@@ -24,10 +26,12 @@ def __init__(self):
         self.client_sampling = actionlib.SimpleActionClient('call_saleae', SaleaeAction)
         self.client_guess = actionlib.SimpleActionClient('guess_acoustics', AcousticsGuessAction)
         self.client_processing = actionlib.SimpleActionClient('process_acoustics', AcousticsProcessingAction)
+        self.client_data = actionlib.SimpleActionClient('generate_data', AcousticsDataAction)
 
         self.client_sampling.wait_for_server()
         self.client_guess.wait_for_server()
         self.client_processing.wait_for_server()
+        self.client_data.wait_for_server()
 
         self.server = actionlib.SimpleActionServer(self.ACTION_NAME, AcousticsWrapperAction, self.execute, False)
```

Comment:
Here, you should read in that ros parameter and create the server to be of type AcousticsWrapperAction or your custom wrapper type for use with data_gen.